samples/system-test/spanner.test.js

```diff
@@ -41,624 +41,705 @@ const spanner = new Spanner({
   projectId: PROJECT_ID,
 });
 
-test.before(tools.checkCredentials);
-
-test.before(async () => {
-  const instance = spanner.instance(INSTANCE_ID);
-  const database = instance.database(DATABASE_ID);
-
-  try {
-    await instance.delete();
-  } catch (err) {
-    // Ignore error
-  }
-
-  try {
-    await database.delete();
-  } catch (err) {
-    // Ignore error
-  }
-
-  const [, operation] = await instance.create({
-    config: 'regional-us-central1',
-    nodes: 1,
-    labels: {
-      'gcloud-sample-tests': 'true',
-    },
+describe('Spanner', () => {
+  before(async () => {
+    tools.checkCredentials();
+    const instance = spanner.instance(INSTANCE_ID);
+    const database = instance.database(DATABASE_ID);
+
+    try {
+      await instance.delete();
+    } catch (err) {
+      // Ignore error
+    }
+
+    try {
+      await database.delete();
+    } catch (err) {
+      // Ignore error
+    }
+
+    const [, operation] = await instance.create({
+      config: 'regional-us-central1',
+      nodes: 1,
+      labels: {
+        'gcloud-sample-tests': 'true',
+      },
+    });
+
+    try {
+      await operation.promise();
+    } catch (err) {
+      // Ignore error
+    }
+
+    const [instances] = await spanner.getInstances({
+      filter: 'labels.gcloud-sample-tests:true',
+    });
+
+    instances.forEach(async instance => {
+      const {operations} = await getOperations(instance.metadata.name);
+
+      operations
```

Comment:
you need to `await Promise.all(operations.filter....)` this line. The `forEach` is returning a list of async functions that need to be await-ed-ed :)

Reply:
Thank you Justin for your feedback. I did the change.